sympy/calculus/tests/test_singularities.py

```diff
@@ -80,12 +80,13 @@ def test_is_strictly_decreasing():
         1/(x**2 - 3*x), Interval.Ropen(-oo, Rational(3, 2)))
     assert not is_strictly_decreasing(-x**2, Interval(-oo, 0))
     assert not is_strictly_decreasing(1)
-    assert is_strictly_decreasing(1/(x**2 - 3*x), Interval.open(1.5, 3))
-
+    assert is_strictly_decreasing(1/(x**2 - 3*x), Interval.open(Rational(3,2), 3))
+    assert is_strictly_decreasing(1/(x**2 - 3*x), Interval.open(1.5, 3)) is None
 
 def test_is_monotonic():
     """Test whether is_monotonic returns correct value."""
-    assert is_monotonic(1/(x**2 - 3*x), Interval.open(1.5, 3))
+    assert is_monotonic(1/(x**2 - 3*x), Interval.open(Rational(3,2), 3))
+    assert is_monotonic(1/(x**2 - 3*x), Interval.open(1.5, 3)) is None
```

Comment:
Changes like this make me hesitant to accept that making rationals and floats unequal is actually a good idea (see [this comment](https://github.com/sympy/sympy/pull/20033#issuecomment-710417294)). 1.5 is exactly equal to 3/2 as a floating point number. The answer here is fortunately not wrong, just not as helpful anymore, but one could imagine scenarios where this breaks things in a more serious way.

This is obviously more of a question for https://github.com/sympy/sympy/pull/20033 than here. This PR just makes things consistent, which is good, and the changes are pretty minimal that it wouldn't be a big deal to revert them if we decide to go the other way.

Reply:
Perhaps we can get the best of both worlds in this case. The mathematical question here is whether the expr is "monotonic" on a certain interval and it most certainly is monotonic along the given `Float` interval. So this function should deal with comparing based on "value equivalence" rather than "structural equivalence". So I think 
https://github.com/sympy/sympy/blob/7c0fa071cb3e2f267a6cb9767a093a8ebbf10375/sympy/sets/sets.py#L399-L400
might be a wrong use of `==` and should use a comparison based on value.